pdns/ipcipher.cc

```diff
@@ -59,10 +59,40 @@ static ComboAddress encryptCA6(const ComboAddress& address, const std::string& k
 
   ComboAddress ret = address;
 
+#if OPENSSL_VERSION_MAJOR >= 3
+  auto ctx = std::unique_ptr<EVP_CIPHER_CTX, decltype(&EVP_CIPHER_CTX_free)>(EVP_CIPHER_CTX_new(), &EVP_CIPHER_CTX_free);
+  if (ctx == nullptr) {
+    throw pdns::OpenSSL::error("encryptCA6: Could not initialize cipher context");
+  }
+
+  EVP_CIPHER_CTX_set_padding(ctx.get(), 0);
```

Comment:
Actually this does not work for me (see `test-ipcrypt_cc.cc`), I need to call `EVP_CIPHER_CTX_set_padding(ctx.get(), 0);` *after* `EVP_EncryptInit` (and *after* `EVP_DecryptInit`) otherwise OpenSSL adds padding, messing things up.

Reply:
That is strange. I was testing using `testrunner --run_test=test_ipcrypt_hh` (which is in `test-ipcrypt_cc.cc`), that's how I figured out the need for explicitly setting the padding to 0. For some reason it also breaks for me now.